src/main/javascript/monet.d.ts

```diff
@@ -331,6 +333,69 @@ declare namespace monet {
     var Nil: Nil;
 
     /****************************************************************
+     * NEL
+     */
+
+    interface NEL<T> extends IMonad<T> {
+        /* Inherited from Monad: */
+        bind<V>(fn: (val: T) => NEL<V>): NEL<V>;
+        flatMap<V>(fn: (val: T) => NEL<V>): NEL<V>;
+        chain<V>(fn: (val: T) => NEL<V>): NEL<V>;
+        map<V>(fn: (val: T) => V): NEL<V>;
+        join<V>(): NEL<V>; // if T is NEL<V>
+        takeLeft<V>(m: NEL<V>): NEL<T>;
+        takeRight<V>(m: NEL<V>): NEL<V>;
+
+        /* from CoMonad: */
+        mapTails<V>(fn: (val: NEL<T>) => V): NEL<V>;
+        cobind<V>(fn: (val: NEL<T>) => V): NEL<V>;
+        coflatMap<V>(fn: (val: NEL<T>) => V): NEL<V>;
+        cojoin(): NEL<NEL<T>>; // if T is NEL<V> === .tails()
+
+        /* Inherited from Applicative */
+        // ap<V>(listFn: List<(val: T) => V>): List<V>;
+
+        /* Validation specific */
+        foldLeft<V>(initial: V): (fn: ListFoldLeftFn<T, V>) => V;
+        foldRight<V>(initial: V): (fn: ListFoldRightFn<T, V>) => V;
+        reduceLeft(fn: ListFoldLeftFn<T, T>): T;
+
+        filter(fn: (val: T) => boolean): NEL<T>;
```

Comment:
This signature is a bug - either in typings, or in monet. `NEL(1).filter(even)` cannot return a `NEL`.